prusti-encoder/src/encoders/mir_impure.rs

```diff
@@ -641,7 +657,7 @@ impl<'tcx, 'vir, 'enc> mir::visit::Visitor<'tcx> for EncVisitor<'tcx, 'vir, 'enc
                         match ty.get_enumlike().filter(|_| place_ty.variant_index.is_none()) {
                             Some(el) => {
                                 let discr_expr = self.vcx.mk_field_expr(place_expr, el.as_ref().unwrap().discr);
-                                self.vcx.mk_unfolding_expr(ty.ref_to_pred.apply(self.vcx, [place_expr]), discr_expr)
+                                self.vcx.mk_unfolding_expr(ty.ref_to_pred.apply(self.vcx, [place_expr], Some(self.vcx.mk_wildcard())), discr_expr)
```

Comment:
Should we eventually axiomatise an infinitesimal `read` permission? Or is this kind of unfolding not running into performance problems?

Reply:
I don't think that this old-Prusti solution will work in the new Prusti. I'm pretty sure that we'll actually need to explicitly keep track of the exact read permission left for all shared references (rather than just abstracting it to some infinitesimal amount). Take the following example:
```rust
// requires acc(List(x), p)
// ensures acc(List(x), p/2)
// ensures acc(T(result), p/2) --* acc(List(x), p/2)
fn reborrow(x: &List<T>) -> &T;
```
This is the only ownership spec that I can see working for this function. If one used `read` then how much permission is returned for `List(x)`: it can't be `read` since the permission for `T(result)` needs to come from somewhere and it's infeasible to decompose `List(x)` and say that `result \in refs_in_list(x)` (i.e. use aliasing without a wand).

Reply:
What I mean is that ultimately the predicate for `&(mut)` will be something like:
```
predicate Ref_T(self: Ref, lft: Map[Ref, Perm]) {
  acc(self.ref) && self.ref in lft && (mutable ? lft[self.ref] == full : lft[self.ref] > none)
}
```